first-party/jib-layer-filter-extension-maven/src/main/java/com/google/cloud/tools/jib/maven/extension/layerfilter/JibLayerFilterExtension.java

```diff
@@ -33,19 +33,39 @@
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.apache.maven.project.DefaultDependencyResolutionRequest;
+import org.apache.maven.project.DependencyResolutionException;
+import org.apache.maven.project.DependencyResolutionResult;
+import org.apache.maven.project.ProjectDependenciesResolver;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.util.filter.ScopeDependencyFilter;
 
+@Named
+@Singleton
 public class JibLayerFilterExtension implements JibMavenPluginExtension<Configuration> {
 
   private Map<PathMatcher, String> pathMatchers = new LinkedHashMap<>();
 
+  @Inject private ProjectDependenciesResolver dependencyResolver;
```

Comment:
Is it possible to annotate this with `@VisibleForTesting` and remove the setter below?

Reply:
Yes, it is.